pkg/apis/operator/v1alpha1/catapult_types.go

```diff
@@ -87,9 +87,9 @@ func (s *CatapultStatus) updatePhase() {
 			s.Phase = CatapultPhasePaused
 			return
 		}
-
+	}
+	for _, condition := range s.Conditions {
```

Comment:
why do we need to go twice through `Conditions`?

Reply:
It was something like:
```golang
func (s *CatapultStatus) updatePhase() {

	for _, condition := range s.Conditions {

		if condition.Type == CatapultPaused && condition.Status == ConditionTrue {
			s.Phase = CatapultPhasePaused
			return
		}

		if condition.Type == CatapultReady {

			switch condition.Status {
			case ConditionTrue:
				s.Phase = CatapultPhaseReady
			case ConditionFalse:
				if condition.Reason == CatapultTerminatingReason {
					s.Phase = CatapultPhaseTerminating
				} else {
					s.Phase = CatapultPhaseNotReady
				}
			case ConditionUnknown:
				s.Phase = CatapultPhaseUnknown
			}
		}
	}
}
```
We need to add `s.Phase = CatapultPhaseUnknown` out of the for loop. Otherwise, if both `Paused` and `Ready` conditions are missing, the `Phase` will not be set. So it becomes:
```golang
func (s *CatapultStatus) updatePhase() {

	for _, condition := range s.Conditions {

		if condition.Type == CatapultPaused && condition.Status == ConditionTrue {
			s.Phase = CatapultPhasePaused
			return
		}

		if condition.Type == CatapultReady {

			switch condition.Status {
			case ConditionTrue:
				s.Phase = CatapultPhaseReady
			case ConditionFalse:
				if condition.Reason == CatapultTerminatingReason {
					s.Phase = CatapultPhaseTerminating
				} else {
					s.Phase = CatapultPhaseNotReady
				}
			case ConditionUnknown:
				s.Phase = CatapultPhaseUnknown
			}
		}
	}
+	s.Phase = CatapultPhaseUnknown
}
```
This cannot work, because it will always set the Phase to `Unknown` if Catapult is `Ready`, so we should return when we set the Phase according to the Ready condition:
```golang
func (s *CatapultStatus) updatePhase() {

	for _, condition := range s.Conditions {

		if condition.Type == CatapultPaused && condition.Status == ConditionTrue {
			s.Phase = CatapultPhasePaused
			return
		}

		if condition.Type == CatapultReady {

			switch condition.Status {
			case ConditionTrue:
				s.Phase = CatapultPhaseReady
			case ConditionFalse:
				if condition.Reason == CatapultTerminatingReason {
					s.Phase = CatapultPhaseTerminating
				} else {
					s.Phase = CatapultPhaseNotReady
				}
			case ConditionUnknown:
				s.Phase = CatapultPhaseUnknown
			}
+                      return
		}
	}
+	s.Phase = CatapultPhaseUnknown
```
This still donsn't work, because when we set the `Paused` condition for an Unready component, we will also set the component to `Ready`, so in the `conditions` slice, if the `Ready` condition comes first, we will set the `Phase` to `Ready` and return, but in this case, the Phase should be `Paused`.

That's the reason that we should first go through the `conditions` to check if the component is `Paused`, then check if it is ready.

Reply:
ok, got it.
let's replace the first loop with
```
	if paused, ok := s.GetCondition(KubeCarrierPaused); ok && paused.True() {
		s.Phase = KubeCarrierPhasePaused
		return
	}
```
:) I know that it's also loop through all conditions, but it looks nicer :)

Reply:
Nice, fixed